rust-runtime/aws-smithy-http-server/src/operation/upgrade.rs

```diff
@@ -264,4 +266,38 @@ where
 }
 
 /// A marker struct indicating an [`Operation`] has not been set in a builder.
+///
+/// This does _not_ implement [`Upgradable`] purposely.
 pub struct MissingOperation;
+
+/// A marker struct indicating an [`Operation`] has not been set in a builder.
+///
+/// This does implement [`Upgradable`] but the output [`Service`] of the upgrade is a [`PanicService`], which will
+/// [`panic`] on [`Service::call`].
+pub struct DummyOperation;
+
+/// A [`Service`] which panics on [`Service::call`].
+#[derive(Clone)]
+pub struct PanicService;
+
+impl<B> Service<http::Request<B>> for PanicService {
+    type Response = http::Response<BoxBody>;
+    type Error = Infallible;
+    type Future = Ready<Result<Self::Response, Self::Error>>;
+
+    fn poll_ready(&mut self, _cx: &mut Context<'_>) -> Poll<Result<(), Self::Error>> {
+        Poll::Ready(Ok(()))
+    }
+
+    fn call(&mut self, _req: Request<B>) -> Self::Future {
+        panic!("this operation was not set")
```

Comment:
Will the user be able to identify to which operation the request was sent that they did not set?

Reply:
I'll add a constructor to the `DummyOperation` to pass an operation name to this `call`.